tests/test_trace.py

```diff
@@ -22,6 +25,20 @@ def full_contracts_cache(config):
     shutil.copytree(BASE_CONTRACTS_PATH, destination)
 
 
+@pytest.fixture
+def show_and_get_trace():
+    def f(receipt: ReceiptAPI, method="show_trace") -> List[str]:
+        with open(TEMP_FILE_NAME, "w+") as temp_file:
+            getattr(receipt, method)(file=temp_file)
+
+        with open("temp", "r") as temp_file:
```

Comment:
See strange things with rich output not showing in `capsys` or anywhere in tests.
The only work-around I have discovered is to write to a file, close the file, and then re-open the file all in the test. Then, I can see the output and make assertions.
I added notes here: https://github.com/Textualize/rich/issues/317#issuecomment-1252320329